app/src/main/java/com/lagradost/cloudstream3/extractors/VidSrcTo.kt

```diff
@@ -0,0 +1,64 @@
+package com.lagradost.cloudstream3.extractors
+
+import android.util.Base64
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.lagradost.cloudstream3.SubtitleFile
+import com.lagradost.cloudstream3.amap
+import com.lagradost.cloudstream3.app
+import com.lagradost.cloudstream3.utils.ExtractorApi
+import com.lagradost.cloudstream3.utils.ExtractorLink
+import java.net.URLDecoder
+import javax.crypto.Cipher
+import javax.crypto.spec.SecretKeySpec
+
+class VidSrcTo : ExtractorApi() {
+    override val name = "VidSrcTo"
+    override val mainUrl = "https://vidsrc.to"
+    override val requiresReferer = true
+
+    override suspend fun getUrl(
+            url: String,
+            referer: String?,
+            subtitleCallback: (SubtitleFile) -> Unit,
+            callback: (ExtractorLink) -> Unit
+    ) {
+        val mediaId = app.get(url).document.selectFirst("ul.episodes li a")?.attr("data-id") ?: return
+        val res = app.get("$mainUrl/ajax/embed/episode/$mediaId/sources").parsedSafe<VidsrctoEpisodeSources>() ?: return
+        if (res?.status != 200) return
+        res.result?.amap { source ->
+            val embedRes = app.get("$mainUrl/ajax/embed/source/${source.id}").parsedSafe<VidsrctoEmbedSource>() ?: return@amap
+            val finalUrl = DecryptUrl(embedRes.result.encUrl)
+            when (source.title) {
+                "Vidplay" -> AnyVidplay(finalUrl.substringBefore("/e/")).getUrl(finalUrl, referer, subtitleCallback, callback)

```

Comment:
I did not catch this in the first review, however when looking at the code a bit more, can this at any point cause an infinite loop? If it can, detect it and throw an exception/return. 

We can catch exceptions, but not inf recursion. Can you just do a assert(finalUrl != url) or smth along those lines.